modules/woocommerce-analytics/classes/wp-woocommerce-analytics-universal.php

```diff
@@ -0,0 +1,327 @@
+
+			$item_details = $this->get_item_details( $product );
+
+			$universal_commands[] = "_wca.push( {
+				'_en': 'woocommerceanalytics_product_checkout',
+				'blog_id': '" . esc_js( $blogid ) . "',
+				'pi': '" . esc_js( $item_details['id'] ) . "',
+				'pn': '" . esc_js( $item_details['name'] ) . "',
+				'pc': '" . esc_js( $item_details['category'] ) . "',
+				'pp': '" . esc_js( $item_details['price'] ) . "',
+				'pq': '" . esc_js( $cart_item['quantity'] ) . "',
+				'ui': '" . esc_js( $this->get_user_id() ) . "',
+			} );";
+		}
+
+		wc_enqueue_js( implode( "\r\n", $universal_commands ) );
+	}
+
+	/**
+	 * After the checkout process, fire an event for each item in the order
+	 *
+	 * @param string $order_id Order Id.
+	 */
+	public function order_process( $order_id ) {
+		$order              = wc_get_order( $order_id );
+		$universal_commands = array();
+		$blogid             = Jetpack::get_option( 'id' );
+
+		// loop through products in the order and queue a purchase event.
+		foreach ( $order->get_items() as $order_item_id => $order_item ) {
+			$product = $order->get_product_from_item( $order_item );
+
+			$item_details = $this->get_item_details( $product );
+
+			$universal_commands[] = "_wca.push( {
+				'_en': 'woocommerceanalytics_product_purchase',
+				'blog_id': '" . esc_js( $blogid ) . "',
+				'pi': '" . esc_js( $item_details['id'] ) . "',
+				'pn': '" . esc_js( $item_details['name'] ) . "',
+				'pc': '" . esc_js( $item_details['category'] ) . "',
+				'pp': '" . esc_js( $item_details['price'] ) . "',
+				'pq': '" . esc_js( $order_item->get_quantity() ) . "',
+				'oi': '" . esc_js( $order->get_order_number() ) . "',
+				'ui': '" . esc_js( $this->get_user_id() ) . "',
+			} );";
+		}
+
+		wc_enqueue_js( implode( "\r\n", $universal_commands ) );
+	}
+
+	/**
+	 * Listen for clicks on the "Update Cart" button to know if an item has been removed by
+	 * updating its quantity to zero
+	 */
+	public function remove_from_cart_via_quantity() {
+		$blogid = Jetpack::get_option( 'id' );
+
+		wc_enqueue_js( "
+			jQuery( 'input[name=update_cart]' ).on( 'click', function() {
+				var cartItems = jQuery( '.cart_item' );
+				cartItems.each( function( item ) {
+					var qty = jQuery( this ).find( 'input.qty' );
+					if ( qty && qty.val() === '0' ) {
+						var productID = jQuery( this ).find( '.product-remove a' ).data( 'product_id' );
+						_wca.push( {
+							'_en': 'woocommerceanalytics_remove_from_cart',
+							'blog_id': '" . esc_js( $blogid ) . "',
+							'pi': productID,
+							'ui': '" . esc_js( $this->get_user_id() ) . "',
+						} );
+					}
+				} );
+			} );
+		" );
+	}
+
+	/**
+	 * Get the current user id
+	 *
+	 * @return int
+	 */
+	public function get_user_id() {
+		if ( is_user_logged_in() ) {
+			$blogid = Jetpack::get_option( 'id' );
+			$userid = get_current_user_id();
+			return $blogid . ":" . $userid;
```

Comment:
Just noting that this is different in the `wc-calypso-bridge` version. I do seem to recall some discussion around this before, and I think this is fine as-is here but wanted to just log that it is different than what has been tested on AT sites.